internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -117,39 +122,16 @@ type CreateOpts struct {
 	Client               secret_service.ClientService
 }
 
-type SecretConfig struct {
-	Version         string
-	Type            integrations.IntegrationType
-	IntegrationName string `yaml:"integration_name"`
-	Details         map[string]any
-}
-
-type MongoDBRole struct {
-	RoleName       string `mapstructure:"role_name"`
-	DatabaseName   string `mapstructure:"database_name"`
-	CollectionName string `mapstructure:"collection_name"`
-}
-
-type MongoDBScope struct {
-	Name string `mapstructure:"type"`
-	Type string `mapstructure:"name"`
-}
-
-type AwsAssumeRole struct {
-	RoleArn string `mapstructure:"role_arn"`
+type DetailsInternal struct {
```

Comment:
if its internal should it be exported?
```suggestion
type detailsInternal struct {
```